docs/sftt/sftt-api-spec-new.md

```diff
@@ -34,13 +41,14 @@ If the block id specified is invalid.
 
 `POST api/v1/protected/reservations/complete`
 
-Must be called on a block reserved by the user. Will change the completion date to the current date and time.
+Must be called on a block reserved by the user. `team_id` is the team that the user wants to credit with this block completion. This can always be left `NULL`.
```

Comment:
If instead we credit the block completion to the team that was specified when it was reserved, then we should document that here and make sure we have some way of indicating that to the user because I could see that creating some confusion

Reply:
My thoughts with this were that if a user reserves a block and specifies a team then that whole team is able to complete the block. This was a logical next step from assigning a team a block with no specific user, since then the team would need the ability to complete the block.

Reply:
Huh... that makes sense but I feel like we could get into trouble with that sense we're assigning different meanings for the team_id field in the reserve and complete endpoints. Also, it might make more sense to have a different flow for reserving a block for a team that is separated from the reserving a block for that user....
@jackblanc What do you think?

Reply:
I like the idea of anyone on a team being able to complete a block that is reserved specifically for that team

Reply:
It somewhat has different meanings, I see that... but at the same time, if a reservation record has a team ID that means it's that team's responsibility? If it's completed, it counts towards that team's success?

Reply:
I don't think it'll get much more complicated than that

Reply:
I think the idea is that once the team is specified it's that team's responsibility and counts towards them?

Reply:
How else would we allow a user to complete their teammate's block?

Reply:
So at reservation there's two options:
1. Don't assign it a team, so the user is solely responsible for this block.
2. Assign it to a team, and so the team can also complete this block later.

Then at completion there's two options as well:
1. The user who reserved the block completes the block, and can choose which, if any, team to credit to. Credit goes to reserving user and the team of his choosing.
2. A teammate completes the block. The teammate cannot specify the team, since it was reserved by another user. Credit goes to the user who completes the block, not the reserving user, and their shared team. 

Does that sound logical?